de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/negotiation/ArchiveIncomingProjectNegotiation.java

```diff
@@ -61,6 +64,19 @@ public ArchiveIncomingProjectNegotiation(
             workspace, checksumCache, connectionService, transmitter, receiver);
     }
 
+    @Override
+    protected void setup(IProgressMonitor monitor)
+        throws IOException {
+        archiveTransferListener = new ArchiveTransferListener(
+            ARCHIVE_TRANSFER_ID + getID());
+
+        if (fileTransferManager == null)
+            // FIXME: the logic will try to send this to the remote contact
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Take the required action to fix the issue indicated by this comment. ([squid:S1134](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS1134))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=160&issues=AWTLb7qM6dA5b-W27MG1&open=AWTLb7qM6dA5b-W27MG1)

[ISSUE_KEY]: AWTLb7qM6dA5b-W27MG1